script/validate-data/index.ts

```diff
@@ -65,30 +64,48 @@ async function checkWorkflows(folders: string[]): Promise<WorkflowWithErrors[]>
   return result;
 }
 
-async function checkWorkflow(workflowPath: string, propertiesPath: string): Promise<WorkflowWithErrors> {
-  let workflowErrors: WorkflowWithErrors = {
-    id: workflowPath,
-    errors: []
-  }
-
+async function validateWorkflowProperties(propertiesPath: string) : Promise<string[]> {
   try {
-    const workflowFileContent = await fs.readFile(workflowPath, "utf8");
-    safeLoad(workflowFileContent); // Validate yaml parses without error
-
+    let errors = []
     const propertiesFileContent = await fs.readFile(propertiesPath, "utf8")
     const properties: WorkflowProperties = JSON.parse(propertiesFileContent)
 
     let v = new validator();
     const res = v.validate(properties, propertiesSchema)
-    workflowErrors.errors = res.errors.map(e => e.toString())
-
+    errors = res.errors.map(e => e.toString())
     if (properties.iconName && !properties.iconName.startsWith("octicon")) {
       try {
         await fs.access(`../../icons/${properties.iconName}.svg`)
       } catch (e) {
-        workflowErrors.errors.push(`No icon named ${properties.iconName} found`)
+        errors.push(`No icon named ${properties.iconName} found`)
       }
     }
+    return errors
+  }
+  catch (e) {
+    throw e
+  }
+}
+
+async function checkWorkflow(workflowPath: string, propertiesPath: string): Promise<WorkflowWithErrors> {
+  let workflowErrors: WorkflowWithErrors = {
+    id: workflowPath,
+    errors: []
+  }
+
+  try {
+    workflowErrors.errors = await validateWorkflowProperties(propertiesPath)
+
+    const workflowFileContent = await fs.readFile(workflowPath, "utf8");
+    const workflow = safeLoad(workflowFileContent); // Validate yaml parses without error
+    
+    let workflowValidator = new validator();
+    const workflowSchema = require("./workflow-schema.json");
```

Comment:
How will this schema file be maintained and updated? If the schema goes out of date with what in production, then we will get false negatives. Hence it is very important to define the clear path for maintenance of this file.

cc @andymckay